pkg/metrics/metrics.go

```diff
@@ -0,0 +1,231 @@
+package metrics
+
+import (
+	"errors"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+)
+
+type MetricDefinition struct {
+	Name   string
+	Help   string
+	Type   metricType
+	Labels []string
+}
+
+type PromConnectionInfo struct {
+	Address string
+	Port    int
+}
+
+type Settings struct {
+	PromConnectionInfo
+	Prefix string
+}
+
+type metricType string
+
+const (
+	TypeCounter   metricType = "counter"
+	TypeGauge     metricType = "gauge"
+	TypeHistogram metricType = "histogram"
+)
+
+var allMetrics = []MetricDefinition{}
+
+func defineMetric(name, help string, t metricType, labels ...string) MetricDefinition {
+	def := MetricDefinition{
+		Name:   name,
+		Help:   help,
+		Type:   t,
+		Labels: labels,
+	}
+	allMetrics = append(allMetrics, def)
+	return def
+}
+
+var (
+	hmapEvictionsTotal = defineMetric(
+		"hashmap_evictions_total",
+		"Number of hashmap evictions total",
+		TypeCounter,
+		"operational",
```

Comment:
I think here and below "operational" is also a FLP leftover that can be removed

Reply:
removed